modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/ControlAcceleratorSupport.java

```diff
@@ -175,7 +175,12 @@ private static void doAcceleratorInstall(final List<? extends MenuItem> items, f
                         }
                         if (!menuitem.isDisable()) {
                             if (menuitem instanceof RadioMenuItem) {
-                                ((RadioMenuItem)menuitem).setSelected(!((RadioMenuItem)menuitem).isSelected());
+                                if(((RadioMenuItem)menuitem).getToggleGroup() == null){
```

Comment:
minor: this group insists on adding spaces after "if" and before "{"

Reply:
You can use a pattern variable here to get rid of all the casts.

Reply:
Updated the code to address above comments